dashboard/src/shared/schema.ts

```diff
@@ -156,3 +200,7 @@ export function validate(
   const valid = ajv.validate(schema, yaml.load(values));
   return { valid: !!valid, errors: ajv.errors };
 }
+
+export function isLegacySchema(schema?: JSONSchemaType<any> | any): boolean {
+  return !!schema?.["$schema"] && schema["$schema"] !== "";
```

Comment:
Since the new format is an OpenAPI Schema Object (as per v3.0.X, not 3.1.x) instead of a JSON schema (well, it is a subset of it), it does not include `$schema`, but perhaps it's not the best way to distinguish between them.
Open to suggestions here.